pkg/subnet/public.go

```diff
@@ -0,0 +1,121 @@
+package subnet
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanche-cli/pkg/application"
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/key"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanche-network-runner/utils"
+	"github.com/ava-labs/avalanchego/ids"
+	avago_constants "github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/formatting/address"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+)
+
+type PublicSubnetDeployer struct {
+	LocalSubnetDeployer
+	baseDir     string
+	privKeyPath string
+	network     models.Network
+	log         logging.Logger
+}
+
+func NewPublicSubnetDeployer(app *application.Avalanche, privKeyPath string, network models.Network) *PublicSubnetDeployer {
+	return &PublicSubnetDeployer{
+		LocalSubnetDeployer: *NewLocalSubnetDeployer(app),
+		baseDir:             app.GetBaseDir(),
```

Comment:
why just don't keep the app instead of the app members?

Reply:
I have had other PRs rejected because I was passing the whole object instead of the individual parameters...I think you yourself have made that objection in the past...

I don't know if there is any guideline here...
¯\_(ツ)_/¯